slice/internal/topology/validation.go

```diff
@@ -29,18 +34,48 @@ func AnyAssignment(admission *kueue.Admission) bool {
 
 // AllAssignmentsValid ensures each PodSetAssignment which has a TopologyAssignment
 // defined the TPUSubBlock topology level.
-func AllAssignmentsValid(admission *kueue.Admission) bool {
+func AllAssignmentsValid(admission *kueue.Admission, nodes map[string]corev1.Node) bool {
 	for _, psa := range admission.PodSetAssignments {
 		if psa.TopologyAssignment == nil {
 			continue
 		}
-		if !IsAssignmentValid(psa) {
+		if !IsAssignmentValid(psa, nodes) {
 			return false
 		}
 	}
 	return true
 }
 
-func IsAssignmentValid(psa kueue.PodSetAssignment) bool {
-	return psa.TopologyAssignment != nil && SubblockLevelIndex(psa.TopologyAssignment) != -1
+func IsAssignmentValid(psa kueue.PodSetAssignment, nodes map[string]corev1.Node) bool {
+	if psa.TopologyAssignment == nil {
+		return false
+	}
+
+	if LevelIndex(psa.TopologyAssignment, core.TPUSubBlockLabel) != -1 {
+		return true
+	}
+
+	hostnameLevelIndex := LevelIndex(psa.TopologyAssignment, corev1.LabelHostname)
+	if hostnameLevelIndex == -1 {
+		return false
+	}
+
+	for _, domain := range psa.TopologyAssignment.Domains {
+		if hostnameLevelIndex >= len(domain.Values) {
+			return false
+		}
+		if GetTPUSubBlockLabelValue(domain, hostnameLevelIndex, nodes) == "" {
+			return false
+		}
```

Comment:
What do you think of this for readability?

```
nodeName := domain.Values[hostnameLevelIndex]
GetTPUSubBlockLabelValue(nodes, nodeName)
```

Reply:
Good point!